openhtf/plugs/frontline/bluetooth_sniffer.py

```diff
@@ -0,0 +1,277 @@
+# Copyright 2017 Google Inc. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Classes for controlling Frontline bluetooth sniffers.
+
+This module provides control over a variety of Frontline (now owned by Teledyne
+Lecroy) bluetooth sniffers. For full details on Frontline's sniffer control
+protocol, which this module partially implements, see the reference document:
+
+http://www.fte.com/docs/usermanuals/comprobe%20automation%20server%20protocol.pdf
+"""
+
+import argparse
+import contextlib
+import datetime
+import logging
+import os
+import socket
+import time
+
+from openhtf.util import files
+
+DEFAULT_AUTOMATION_PORT = 22901
+DEFAULT_HTTP_PORT = 55455
+REMOTE_BIN_PATH = (r'C:\Program Files (x86)\Frontline Test System II'
```

Comment:
some docs on these magic values would be good

Reply:
Done.